src/io.c

```diff
@@ -154,13 +155,15 @@ RETSIGTYPE IO_SIGCHLDHandler( int whichsig )
     retcode = waitpid(-1, &status, WNOHANG);
     if (retcode > 0) {   /* One of our child processes terminated */
         if (WIFEXITED(status) || WIFSIGNALED(status)) {
-            if (!statsfull) {
-                stats[lastats] = status;
-                pids[lastats++] = retcode;
-                if (lastats >= maxstats) lastats = 0;
-                if (lastats == fistats) statsfull = 1;
-            } else
-                Pr("#E Overflow in table of terminated processes\n",0,0);
+            if(!CheckChildStatusChanged(retcode, status)) {
```

Comment:
I am somewhat unhappy that this change breaks compatibility with all prior versions of GAP (and indeed, even if the current `master` branch). I'd be more comfortable if this was conditional inside an `#ifdef`.
One way to do that would be to add a check for `CheckChildStatusChanged` to `configure`; another, perhaps more convenient (for package developers) approach would be if GAP added something like `#define GAP_HAVE_CheckChildStatusChanged 1`.

Also: missing space after `if`

Reply:
I tried to think of how to make this work, should have thought of just adding a `define` to GAP! WIll do that.